core/lib/zksync_core/src/api_server/web3/metrics.rs

```diff
@@ -195,3 +195,61 @@ pub(super) struct PubSubMetrics {
 
 #[vise::register]
 pub(super) static PUB_SUB_METRICS: vise::Global<PubSubMetrics> = vise::Global::new();
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EncodeLabelValue, EncodeLabelSet)]
+#[metrics(label = "filter_type", rename_all = "snake_case")]
+pub(super) enum FilterType {
+    Events,
+    Blocks,
+    PendingTransactions,
+}
+
+impl From<TypedFilter> for FilterType {
+    fn from(value: TypedFilter) -> Self {
+        match value {
+            TypedFilter::Events(_, _) => FilterType::Events,
+            TypedFilter::Blocks(_) => FilterType::Blocks,
+            TypedFilter::PendingTransactions(_) => FilterType::PendingTransactions,
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct InstalledFilter {
+    pub filter: TypedFilter,
+    guard: Box<GaugeGuard>,
+    start: Instant,
```

Comment:
Bikeshedding: Please use more informative field names, such as `created_at` in this case.